src/main/kotlin/uk/gov/justice/digital/hmpps/authorizationserver/config/AuthorizationServerConfig.kt

```diff
@@ -39,19 +43,38 @@ class AuthorizationServerConfig(
   @Value("\${jwt.jwk.key.id}") private val keyId: String,
   @Value("\${server.base-url}") private val baseUrl: String,
   @Value("\${server.servlet.context-path}") private val contextPath: String,
+
+  private val clientConfigRepository: ClientConfigRepository,
+  private val ipAddressHelper: IpAddressHelper,
 ) {
 
   @Bean
   @Order(Ordered.HIGHEST_PRECEDENCE)
   fun authorizationServerSecurityFilterChain(http: HttpSecurity): SecurityFilterChain {
-    OAuth2AuthorizationServerConfiguration.applyDefaultSecurity(http.cors().and())
+    val authorizationServerConfigurer = OAuth2AuthorizationServerConfigurer()
+    http.apply(authorizationServerConfigurer)
+    authorizationServerConfigurer.tokenEndpoint { tokenEndpointConfigurer ->
+      tokenEndpointConfigurer.authenticationProviders {
+          authenticationProviders ->
+        authenticationProviders.replaceAll { authenticationProvider -> withIPAddressCheckForClientCredentials(authenticationProvider) }
+      }
+    }
 
-    http.cors().and().csrf().disable()
-      .formLogin(withDefaults<FormLoginConfigurer<HttpSecurity>>())
+    // TODO - confirm cors and csrf configuration
```

Comment:
HAAR-1497 raised to cover confirming CORS and CSRF configuration (plus update any other deprecated CORS/CSRF configuration)